src/content/1.8/code/agda/snippet01.agda

```diff
@@ -0,0 +1,5 @@
+record Bifunctor (F : Set → Set → Set) : Set₁ where
+  field
+    bimap : (A → C) → (B → D) → F A B → F C D
+    first : (A → C) → F A B → F C B
```

Comment:
Closer to the Haskell implementation would be to provide implementations for first and second, something like:
```agda
record Bifunctor (F : Set → Set → Set) : Set₁ where
  field bimap : (A → C) → (B → D) → F A B → F C D
  first : (A → C) → F A B → F C B
  first = {!   !}
  second : (B → D) → F A B → F A D
  second = {!   !}
```
and fill the holes